component/background/StorageService.tsx

```diff
@@ -5,24 +5,46 @@ const STORAGE_TYPE = "sync";
 
 class StorageService {
   private storage: chrome.storage.StorageArea;
+  private status: boolean;
 
   constructor() {
     this.storage = chrome.storage.sync;
-    this.storage.get(STORAGE_KEY_ID, (response) => {
-      // If user id is not locally stored, generate a new one.
-      if (!response[STORAGE_KEY_ID]) {
-        const data: any = {};
-        data[STORAGE_KEY_ID] = this.generateUUID();
-        this.storage.set(data);
-      }
-    });
+    this.status = false;
+  }
+
+  /**
+   * Initialize the IStorageService
+   * @return {Promise<any>} A promise whose resolve takes no argument, which runs after the service is fully initialized.
+   */
+  public initialize(): Promise<any> {
+    if (this.status) {
+      return Promise.resolve();
+    } else {
+      return new Promise((resolve) => {
+        this.storage.get(STORAGE_KEY_ID, (response) => {
+          // If user id is not locally stored, generate a new one.
+          if (!response[STORAGE_KEY_ID]) {
+            const data: any = {};
+            data[STORAGE_KEY_ID] = this.generateUUID();
+            this.storage.set(data);
```

Comment:
You should pass resolve, as the second argument like
`this.storage.set(data, resolve);`